processor/src/main/java/org/mapstruct/ap/internal/model/NestedTargetPropertyMappingHolder.java

```diff
@@ -229,6 +233,13 @@ public NestedTargetPropertyMappingHolder build() {
                         handledTargets.add( entryByTP.getKey().getName() );
                     }
 
+                    handleSourceParameterMappings(
```

Comment:
so, IIUC, if no nesting level remains on source side, you assume the source parameter to be mapped directly?